src/sipmfit.jl

```diff
@@ -106,9 +105,33 @@ function fit_sipm_spectrum(pe_cal::Vector{<:Real}, min_pe::Real=0.5, max_pe::Rea
     get_pe_pos = pe -> let sel = in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)
         dot(view(μ, sel), view(w, sel)) / sum(view(w, sel))
     end
+
+    # get pe resolution (FWHM)
     get_pe_res = pe -> let sel = in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)
-        sqrt(dot(view(σ, sel).^2, view(w, sel).^2))
-    end 
+        model_parameters = [
+            view(μ_ml, sel),
+            view(σ_ml, sel),
+            view(w_ml, sel)
+        ]    
+        
+        # Find maximum of PDF in the expected range, step size depends on the maximum variance of the single gaussians
+        range_steps = maximum([μ[i].err for i in size(μ)]) / 100
```

Comment:
```suggestion
        range_steps = maximum(getfield.(μ, :err)) / 100
```
Check if this works. If not, use eachindex instead of size